app/views/participants/_participant.html.erb

```diff
@@ -1,19 +1,35 @@
 <% userid << participant.user_id %>
+
 <tr class= "exp">
-  <td ><%= link_to User.find(participant.user_id).name, impersonate_impersonate_path(:user => {:name => User.find(participant.user_id).name}), :method => :post %></td>
-  <td ><%= link_to User.find(participant.user_id).fullname, :controller=> 'users', :action => 'show', :id => participant.user_id %></td>
+  <td><%= link_to User.find(participant.user_id).name, impersonate_impersonate_path(:user => {:name => User.find(participant.user_id).name}), :method => :post %></td>
+  <td><%= link_to User.find(participant.user_id).fullname, :controller=> 'users', :action => 'show', :id => participant.user_id %></td>
   <td><a href="mailto: <%= User.find(participant.user_id).email %>"><%= User.find(participant.user_id).email %></a></td>
-  <td ><%= User.find(participant.user_id).role.name %></td>
-  <td  align="center"><%= (User.find(User.find(participant.user_id).parent_id)).name %></td>
-  <td align="center" ><%= User.yesorno(User.find(participant.user_id).email_on_review) %></td>
-  <td align="center" ><%= User.yesorno(User.find(participant.user_id).email_on_submission) %></td>
-  <td align="center" ><%= User.yesorno(User.find(participant.user_id).email_on_review_of_review) %></td>
-  <td align="center" ><%= participant.can_submit == false ? 'no' : 'yes' %></td>
-  <td align="center" ><%= participant.can_review == false ? 'no' : 'yes' %></td>
-  <td align="center" ><%= participant.can_take_quiz == false ? 'no' : 'yes' %></td>
-  <td align="center" ><%= participant.handle %></td>
-
-  <td align="center" ><%= link_to 'Remove', {:controller => controller, :action => 'destroy', :id => participant.id}, :method => :delete %></td>
+  <td><%= User.find(participant.user_id).role.name %></td>
+  <td align="center"><%= (User.find(User.find(participant.user_id).parent_id)).name %></td>
+
+  <% if @model == 'Assignment' %>
```

Comment:
Very good! Currently, Expertiza only supports badges for assignment participants, instead of course participants.